File: PolicyPlus/ImportSpol.vb
Public Class ImportSpol
    Public Spol As SpolFile
    Private Sub ButtonOpenFile_Click(sender As Object, e As EventArgs) Handles ButtonOpenFile.Click
        Using ofd As New OpenFileDialog
            ofd.Filter = "Semantic Policy files|*.spol|All files|*.*"
            If ofd.ShowDialog = DialogResult.OK Then
                TextSpol.Text = IO.File.ReadAllText(ofd.FileName)
            End If
        End Using
    End Sub
    Private Sub ButtonVerify_Click(sender As Object, e As EventArgs) Handles ButtonVerify.Click
        Try
            Dim spol = SpolFile.FromText(TextSpol.Text)
            MsgBox("Validation successful, " & spol.Policies.Count & " policy settings found.", MsgBoxStyle.Information)
        Catch ex As Exception
            MsgBox("SPOL validation failed: " & ex.Message, MsgBoxStyle.Exclamation)
        End Try
    End Sub
    Private Sub ButtonApply_Click(sender As Object, e As EventArgs) Handles ButtonApply.Click
        Try
            Spol = SpolFile.FromText(TextSpol.Text) ' Tell the main form that the SPOL is ready to be committed
            DialogResult = DialogResult.OK
        Catch ex As Exception
            MsgBox("The SPOL text is invalid: " & ex.Message, MsgBoxStyle.Exclamation)
        End Try
    End Sub
    Private Sub ImportSpol_Shown(sender As Object, e As EventArgs) Handles Me.Shown
        Spol = Nothing
    End Sub
    Private Sub ImportSpol_KeyUp(sender As Object, e As KeyEventArgs) Handles Me.KeyUp
        If e.KeyCode = Keys.Escape And Not (TextSpol.Focused And TextSpol.SelectionLength > 0) Then DialogResult = DialogResult.Cancel
    End Sub
    Private Sub TextSpol_KeyDown(sender As Object, e As KeyEventArgs) Handles TextSpol.KeyDown
        If e.KeyCode = Keys.A And e.Control Then TextSpol.SelectAll()
    End Sub
    Private Sub ButtonReset_Click(sender As Object, e As EventArgs) Handles ButtonReset.Click
        If MsgBox("Are you sure you want to reset the text box?", MsgBoxStyle.Question Or MsgBoxStyle.YesNo) = MsgBoxResult.Yes Then
            TextSpol.Text = "Policy Plus Semantic Policy" & vbCrLf & vbCrLf
        End If
    End Sub
End Class

File: PolicyPlus/ImportSpol.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace PolicyPlus
{
    public partial class ImportSpol
    {
        public SpolFile Spol;

        public ImportSpol()
        {
            InitializeComponent();
        }
        private void ButtonOpenFile_Click(object sender, EventArgs e)
        {
            using (var ofd = new OpenFileDialog())
            {
                ofd.Filter = "Semantic Policy files|*.spol|All files|*.*";
                if (ofd.ShowDialog() == DialogResult.OK)
                {
                    TextSpol.Text = System.IO.File.ReadAllText(ofd.FileName);
                }
            }
        }
        private void ButtonVerify_Click(object sender, EventArgs e)
        {
            try
            {
                var spol = SpolFile.FromText(TextSpol.Text);
                Interaction.MsgBox("Validation successful, " + spol.Policies.Count + " policy settings found.", MsgBoxStyle.Information);
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("SPOL validation failed: " + ex.Message, MsgBoxStyle.Exclamation);
            }
        }
        private void ButtonApply_Click(object sender, EventArgs e)
        {
            try
            {
                Spol = SpolFile.FromText(TextSpol.Text); // Tell the main form that the SPOL is ready to be committed
                DialogResult = DialogResult.OK;
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("The SPOL text is invalid: " + ex.Message, MsgBoxStyle.Exclamation);
            }
        }
        private void ImportSpol_Shown(object sender, EventArgs e)
        {
            Spol = null;
        }
        private void ImportSpol_KeyUp(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape & !(TextSpol.Focused & TextSpol.SelectionLength > 0))
                DialogResult = DialogResult.Cancel;
        }
        private void TextSpol_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.A & e.Control)
                TextSpol.SelectAll();
        }
        private void ButtonReset_Click(object sender, EventArgs e)
        {
            if (Interaction.MsgBox("Are you sure you want to reset the text box?", MsgBoxStyle.Question | MsgBoxStyle.YesNo) == MsgBoxResult.Yes)
            {
                TextSpol.Text = "Policy Plus Semantic Policy" + Constants.vbCrLf + Constants.vbCrLf;
            }
        }
    }
}